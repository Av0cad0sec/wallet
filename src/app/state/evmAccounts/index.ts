import { PayloadAction } from '@reduxjs/toolkit'
import { createSlice } from 'utils/@reduxjs/toolkit'
import { EvmAccounts } from './types'

export const initialState: EvmAccounts = {}

const slice = createSlice({
  name: 'evmAccounts',
  initialState,
  reducers: {
  },
})

export const { actions: evmAccountsActions } = slice

export default slice.reducer
