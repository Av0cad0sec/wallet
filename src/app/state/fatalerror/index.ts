import { PayloadAction } from '@reduxjs/toolkit'
import { createSlice } from 'utils/@reduxjs/toolkit'
import { useInjectReducer } from 'utils/redux-injectors'
import { FatalErrorPayload, FatalErrorState } from './types'

export const initialState: FatalErrorState = {}

const slice = createSlice({
  name: 'fatalError',
  initialState,
  reducers: {
    setError(state, action: PayloadAction<FatalErrorPayload>) {
      if (!state.error) {
        state.error = action.payload
      }
    },
  },
})

export const { actions: fatalErrorActions } = slice

export const useFatalErrorSlice = () => {
  useInjectReducer({ key: slice.name, reducer: slice.reducer })
  return { actions: slice.actions }
}
