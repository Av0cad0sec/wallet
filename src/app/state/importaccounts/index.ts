import { PayloadAction } from '@reduxjs/toolkit'
import { ErrorPayload } from 'types/errors'
import { createSlice } from 'utils/@reduxjs/toolkit'
import { ImportAccountsListAccount, ImportAccountsState, ImportAccountsStep } from './types'

export const initialState: ImportAccountsState = {
  accounts: [],
  showAccountsSelectionModal: false,
  step: ImportAccountsStep.Idle,
}

const slice = createSlice({
  name: 'importAccounts',
  initialState,
  reducers: {
    clear(state) {
      state.accounts = []
      state.error = undefined
      state.step = ImportAccountsStep.Idle
      state.showAccountsSelectionModal = false
    },
    enumerateAccountsFromLedger(state) {
      state.accounts = []
      state.showAccountsSelectionModal = true
      state.step = ImportAccountsStep.Idle
    },
    enumerateAccountsFromMnemonic(state, _action: PayloadAction<string>) {
      state.step = ImportAccountsStep.Idle
      state.accounts = []
      state.showAccountsSelectionModal = true
    },
    toggleAccount(state, action: PayloadAction<number>) {
      const index = action.payload
      state.accounts[index].selected = !state.accounts[index].selected
    },
    accountsListed(state, action: PayloadAction<ImportAccountsListAccount[]>) {
      state.accounts = action.payload
    },
    setStep(state, action: PayloadAction<ImportAccountsStep>) {
      state.step = action.payload
    },
    operationFailed(state, action: PayloadAction<ErrorPayload>) {
      state.error = action.payload
      state.step = ImportAccountsStep.Idle
    },
  },
})

export const { actions: importAccountsActions } = slice

export default slice.reducer
