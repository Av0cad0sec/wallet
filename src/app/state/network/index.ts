import { PayloadAction } from '@reduxjs/toolkit'
import { createSlice } from 'utils/@reduxjs/toolkit'
import { useInjectReducer, useInjectSaga } from 'utils/redux-injectors'
import { networkSaga } from './saga'
import { NetworkState, NetworkType } from './types'

export const initialState: NetworkState = {
  ticker: '',
  chainContext: '',
  selectedNetwork: 'local',
}

const slice = createSlice({
  name: 'network',
  initialState,
  reducers: {
    selectNetwork(state, action: PayloadAction<NetworkType>) {},
    networkSelected(state, action: PayloadAction<NetworkState>) {
      Object.assign(state, action.payload)
    },
  },
})

export const { actions: networkActions } = slice

export const useNetworkSlice = () => {
  useInjectReducer({ key: slice.name, reducer: slice.reducer })
  useInjectSaga({ key: slice.name, saga: networkSaga })
  return { actions: slice.actions }
}
