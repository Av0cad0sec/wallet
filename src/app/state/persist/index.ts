import { PayloadAction } from '@reduxjs/toolkit'
import { createSlice } from 'utils/@reduxjs/toolkit'
import { PersistState } from './types'
import { runtimeIs } from 'config'

// eslint-disable-next-line @typescript-eslint/no-unused-vars
import { createPersistedRootReducer } from 'store/reducers'
import { RootState } from 'types'

export const STORAGE_FIELD = 'oasis_wallet_persist_v1'

/** Syncing tabs is only needed in web app, not in extension. */
export const needsSyncingTabs = runtimeIs === 'webapp'

// Simulate with `delete window.BroadcastChannel`
export const isSyncingTabsSupported = typeof BroadcastChannel === 'function'

export function getInitialState(): PersistState {
  return {
    hasPersistedProfiles: !!localStorage.getItem(STORAGE_FIELD),
    // Disable persistence if tabs would override each other.
    isPersistenceUnsupported: needsSyncingTabs && !isSyncingTabsSupported,
    loading: false,
    stringifiedEncryptionKey: undefined,
    enteredWrongPassword: false,
  }
}

const persistSlice = createSlice({
  name: 'persist',
  initialState: getInitialState(),
  reducers: {
    setWrongPassword(state) {
      state.enteredWrongPassword = true
      state.loading = false
    },
    setUnlockedRootState(state, action: PayloadAction<{ rootState: RootState }>) {
      /**
       * Handled in {@link createPersistedRootReducer}.
       * Sets `state.loading = false` and `state.stringifiedEncryptionKey`.
       */
      return
    },

    // Handled in saga
    // ---------------
    /**
     * Set a password to encrypt state with and begin storing it to localStorage.
     */
    setPasswordAsync(state, action: PayloadAction<{ password: string }>) {
      state.loading = true
    },
    /**
     * Load encrypted state from localStorage and decrypt, or set {@link PersistState.enteredWrongPassword}.
     */
    unlockAsync(state, action: PayloadAction<{ password: string }>) {
      state.loading = true
    },
    /**
     * Remove everything from state, especially {@link PersistState.stringifiedEncryptionKey}.
     * UnlockGate should stop any components relying on missing state.
     */
    lockAsync(state) {
      state.loading = true
    },
    /**
     * Remove encrypted state from localStorage and reload.
     */
    eraseAsync(state) {
      state.loading = true
    },
  },
})

export const persistActions = persistSlice.actions

export default persistSlice.reducer
