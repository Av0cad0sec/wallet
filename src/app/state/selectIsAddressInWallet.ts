import { createSelector } from '@reduxjs/toolkit'
import { selectAccountAddress } from 'app/state/account/selectors'
import { selectAddress, selectIsOpen } from 'app/state/wallet/selectors'

export const selectIsAddressInWallet = createSelector(
  [selectIsOpen, selectAddress, selectAccountAddress],
  (walletIsOpen, walletAddress, address) => walletIsOpen && walletAddress === address,
)
