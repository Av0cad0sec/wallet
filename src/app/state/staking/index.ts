import { PayloadAction } from '@reduxjs/toolkit'
import { ErrorPayload } from 'types/errors'
import { createSlice } from 'utils/@reduxjs/toolkit'

import { DebondingDelegation, Delegation, StakingState, Validators, ValidatorDetails } from './types'

export const initialState: StakingState = {
  debondingDelegations: [],
  delegations: [],
  updateDelegationsError: undefined,
  validators: null,
  updateValidatorsError: null,
  selectedValidatorDetails: null,
  selectedValidator: null,
  loading: false,
}

const slice = createSlice({
  name: 'staking',
  initialState,
  reducers: {
    setLoading(state, action: PayloadAction<boolean>) {
      state.loading = action.payload
    },
    fetchAccount(state, action: PayloadAction<string>) {},
    updateValidators(state, action: PayloadAction<Validators>) {
      state.updateValidatorsError = null
      state.validators = action.payload
    },
    updateValidatorsError(state, action: PayloadAction<{ error: string; validators: Validators }>) {
      state.updateValidatorsError = action.payload.error
      state.validators = action.payload.validators
    },
    updateDelegations(
      state,
      action: PayloadAction<{ delegations: Delegation[]; debondingDelegations: DebondingDelegation[] }>,
    ) {
      state.updateDelegationsError = undefined
      state.delegations = action.payload.delegations
      state.debondingDelegations = action.payload.debondingDelegations
    },
    updateDelegationsError(state, action: PayloadAction<ErrorPayload>) {
      state.updateDelegationsError = action.payload
    },
    validatorSelected(state, action: PayloadAction<string>) {
      state.selectedValidator = action.payload
      state.selectedValidatorDetails = null
    },
    validatorDeselected(state, action: PayloadAction<void>) {
      state.selectedValidator = null
      state.selectedValidatorDetails = null
    },
    updateValidatorDetails(state, action: PayloadAction<ValidatorDetails>) {
      state.selectedValidatorDetails = action.payload
    },
  },
})

export const { actions: stakingActions, reducer: stakingReducer } = slice

export default slice.reducer
