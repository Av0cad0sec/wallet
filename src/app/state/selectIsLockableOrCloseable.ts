import { createSelector } from '@reduxjs/toolkit'
import { selectUnlockedStatus } from 'app/state/selectUnlockedStatus'
import { ExhaustedTypeError } from 'types/errors'

export const selectIsLockableOrCloseable = createSelector([selectUnlockedStatus], unlockedStatus => {
  if (unlockedStatus === 'skippedUnlockingProfile') return 'closeable'
  if (unlockedStatus === 'unlockedProfile') return 'lockable'
  if (unlockedStatus === 'openUnpersisted') return 'closeable'
  if (unlockedStatus === 'lockedProfile') return false
  if (unlockedStatus === 'emptyUnpersisted') return false
  throw new ExhaustedTypeError('Invalid unlocked status', unlockedStatus)
})
