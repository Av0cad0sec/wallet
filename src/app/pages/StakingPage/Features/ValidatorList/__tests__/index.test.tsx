import { render, screen, waitFor } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import { initialState, stakingActions } from 'app/state/staking'
import * as React from 'react'
import { Provider } from 'react-redux'
import { configureAppStore } from 'store/configureStore'

import { ValidatorList } from '..'

jest.mock('react-i18next', () => ({
  useTranslation: () => {
    return {
      t: str => str,
      i18n: {
        changeLanguage: () => new Promise(() => {}),
      },
    }
  },
}))

const renderComponent = store =>
  render(
    <Provider store={store}>
      <ValidatorList />
    </Provider>,
  )

describe('<ValidatorList  />', () => {
  let store: ReturnType<typeof configureAppStore>

  beforeEach(() => {
    store = configureAppStore({ staking: initialState })
  })

  it('should match snapshot', () => {
    store.dispatch(stakingActions.updateValidators([]))
    const component = renderComponent(store)
    expect(component.container.firstChild).toMatchSnapshot()
  })

  it('should display validator details on click', async () => {
    store = configureAppStore({
      staking: {
        ...initialState,
        validators: [
          {
            address: 'oasis123456789',
            fee: 10,
            rank: 0,
            status: 'active',
            escrow: 1000,
            name: 'test-validator',
            media: {
              email_address: 'test@test.com',
              tg_chat: 'telegram',
              twitter_acc: 'https://twitter.com/my_twitter',
              website_link: 'https://test.com',
            },
          },
        ],
      },
    })

    renderComponent(store)

    let row = screen.getByText(/test-validator/)
    expect(row).toBeVisible()
    userEvent.click(row)

    const details = screen.getByTestId('validator-item')
    row = screen.getAllByText(/test-validator/)[0]

    expect(details).toBeVisible()
    userEvent.click(row)
    await waitFor(() => expect(details).not.toBeVisible())
  })

  it('should only display the details of a single validator', async () => {
    store = configureAppStore({
      staking: {
        ...initialState,
        validators: [
          {
            address: 'oasis1validator1',
            fee: 10,
            rank: 0,
            status: 'active',
            escrow: 1000,
            name: 'test-validator1',
            media: {
              email_address: 'test@test.com',
              tg_chat: 'telegram',
              twitter_acc: 'https://twitter.com/my_twitter',
              website_link: 'https://test.com',
            },
          },
          {
            address: 'oasis1validator2',
            fee: 10,
            rank: 1,
            status: 'inactive',
            escrow: 1000,
            name: 'test-validator2',
            media: {
              email_address: 'test@test.com',
              tg_chat: 'telegram',
              twitter_acc: 'https://twitter.com/my_twitter',
              website_link: 'https://test.com',
            },
          },
        ],
      },
    })

    renderComponent(store)

    let row = screen.getByText(/test-validator1/)
    expect(row).toBeVisible()
    userEvent.click(row)

    let details = screen.getByTestId('validator-item')
    expect(screen.getByTestId('validator-item-name').textContent).toEqual('test-validator1')
    expect(details).toBeVisible()

    row = screen.getByText(/test-validator2/)
    userEvent.click(row)

    details = screen.getByTestId('validator-item')
    expect(screen.getByTestId('validator-item-name').textContent).toEqual('test-validator2')
    expect(details).toBeVisible()
  })
})
