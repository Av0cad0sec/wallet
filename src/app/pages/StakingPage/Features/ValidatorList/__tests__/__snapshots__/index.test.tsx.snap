// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ValidatorList  /> empty should match snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  background-color: #FFFFFF;
  color: #444444;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 24px;
}

.c1 {
  margin-top: 0px;
  font-size: 34px;
  line-height: 40px;
  max-width: 816px;
  font-weight: 600;
  overflow-wrap: break-word;
}

.c4 {
  position: relative;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
  height: 100%;
  max-width: 100%;
  color: false;
  background-color: false;
}

.c2 {
  position: relative;
  width: 100%;
  border-radius: inherit;
  overflow-x: auto;
  overflow-y: hidden;
  min-height: 0;
}

.c3 {
  position: relative;
  width: 100%;
  display: table;
}

.c5 {
  box-sizing: border-box;
  width: 100%;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  color: false;
  background-color: false;
}

@media only screen and (max-width:768px) {
  .c0 {
    padding: 12px;
  }
}

@media only screen and (max-width:768px) {
  .c1 {
    margin-top: 0px;
  }
}

@media only screen and (max-width:768px) {
  .c1 {
    font-size: 26px;
    line-height: 32px;
    max-width: 624px;
  }
}

<div
  class="c0"
>
  <h1
    class="c1"
  >
    common.validators
  </h1>
  <div
    class="c2"
  >
    <div
      class="c3"
    >
      <div
        class="c4 rdt_Table"
        role="table"
      >
        <div
          class="c5"
        >
          <div
            style="padding: 24px;"
          >
            There are no records to display
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<ValidatorList  /> list should match snapshot 1`] = `
.c11 {
  display: inline-block;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  width: 24px;
  height: 24px;
  fill: #666666;
  stroke: #666666;
}

.c11 g {
  fill: inherit;
  stroke: inherit;
}

.c11 *:not([stroke])[fill="none"] {
  stroke-width: 0;
}

.c11 *[stroke*="#"],.c11 *[STROKE*="#"] {
  stroke: inherit;
  fill: none;
}

.c11 *[fill-rule],
.c11 *[FILL-RULE],
.c11 *[fill*="#"],.c11 *[FILL*="#"] {
  fill: inherit;
  stroke: none;
}

.c18 {
  display: inline-block;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  width: 24px;
  height: 24px;
  fill: #00C781;
  stroke: #00C781;
}

.c18 g {
  fill: inherit;
  stroke: inherit;
}

.c18 *:not([stroke])[fill="none"] {
  stroke-width: 0;
}

.c18 *[stroke*="#"],.c18 *[STROKE*="#"] {
  stroke: inherit;
  fill: none;
}

.c18 *[fill-rule],
.c18 *[FILL-RULE],
.c18 *[fill*="#"],.c18 *[FILL*="#"] {
  fill: inherit;
  stroke: none;
}

.c20 {
  display: inline-block;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  width: 24px;
  height: 24px;
  fill: #FF4040;
  stroke: #FF4040;
}

.c20 g {
  fill: inherit;
  stroke: inherit;
}

.c20 *:not([stroke])[fill="none"] {
  stroke-width: 0;
}

.c20 *[stroke*="#"],.c20 *[STROKE*="#"] {
  stroke: inherit;
  fill: none;
}

.c20 *[fill-rule],
.c20 *[FILL-RULE],
.c20 *[fill*="#"],.c20 *[FILL*="#"] {
  fill: inherit;
  stroke: none;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  background-color: #FFFFFF;
  color: #444444;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 24px;
}

.c19 {
  margin-left: 3px;
  font-size: 18px;
  line-height: 24px;
}

.c1 {
  margin-top: 0px;
  font-size: 34px;
  line-height: 40px;
  max-width: 816px;
  font-weight: 600;
  overflow-wrap: break-word;
}

.c4 {
  position: relative;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
  height: 100%;
  max-width: 100%;
  color: false;
  background-color: false;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  text-align: left;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  width: 100%;
  background-color: false;
  min-height: 56px;
  border-bottom-width: 1px;
  border-bottom-color: #AAAAAAaa;
  border-bottom-style: solid;
}

.c7 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  line-height: normal;
  font-size: 18px;
  font-weight: 500;
  color: false;
  padding-left: 16px;
  padding-right: 16px;
}

.c16 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  line-height: normal;
  padding-left: 16px;
  padding-right: 16px;
  word-break: break-word;
}

.c8 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-flex-basis: 0;
  -ms-flex-preferred-size: 0;
  flex-basis: 0;
  max-width: 100%;
  min-width: 100px;
  min-width: 34px;
  max-width: 34px;
}

.c9 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-flex-basis: 0;
  -ms-flex-preferred-size: 0;
  flex-basis: 0;
  max-width: 100%;
  min-width: 100px;
}

.c12 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-flex-basis: 0;
  -ms-flex-preferred-size: 0;
  flex-basis: 0;
  max-width: 100%;
  min-width: 100px;
}

.c13 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-flex-basis: 0;
  -ms-flex-preferred-size: 0;
  flex-basis: 0;
  max-width: 100%;
  min-width: 100px;
  min-width: 110px;
  max-width: 110px;
}

.c17 {
  font-weight: 400;
}

.c15 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  width: 100%;
  box-sizing: border-box;
  font-size: 16x;
  color: false;
  background-color: false;
  min-height: 48px;
}

.c15:not(:last-of-type) {
  border-bottom-style: solid;
  border-bottom-width: 1px;
  border-bottom-color: #AAAAAAaa;
}

.c15:hover {
  color: false;
  background-color: #88888833;
  -webkit-transition-duration: 0.15s;
  transition-duration: 0.15s;
  -webkit-transition-property: background-color;
  transition-property: background-color;
  border-bottom-color: false;
  outline-style: solid;
  outline-width: 1px;
  outline-color: false;
}

.c15:hover {
  cursor: pointer;
}

.c10 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 100%;
  line-height: 1;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c10:focus {
  outline: none;
  color: false;
}

.c10:hover {
  color: false;
}

.c10 span.__rdt_custom_sort_icon__ i,
.c10 span.__rdt_custom_sort_icon__ svg {
  opacity: 0;
  color: inherit;
  font-size: 18px !important;
  height: 18px !important;
  width: 18px !important;
  -webkit-backface-visibility: hidden;
  backface-visibility: hidden;
  -webkit-transform-style: preserve-3d;
  -ms-transform-style: preserve-3d;
  transform-style: preserve-3d;
  -webkit-transition-duration: 125ms;
  transition-duration: 125ms;
  -webkit-transition-property: -webkit-transform;
  -webkit-transition-property: transform;
  transition-property: transform;
}

.c10 span.__rdt_custom_sort_icon__.asc i,
.c10 span.__rdt_custom_sort_icon__.asc svg {
  -webkit-transform: rotate(180deg);
  -ms-transform: rotate(180deg);
  transform: rotate(180deg);
}

.c10:hover {
  cursor: pointer;
}

.c10:hover span,
.c10:hover span.__rdt_custom_sort_icon__ * {
  opacity: 1;
}

.c14 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c2 {
  position: relative;
  width: 100%;
  border-radius: inherit;
  overflow-x: auto;
  overflow-y: hidden;
  min-height: 0;
}

.c3 {
  position: relative;
  width: 100%;
  display: table;
}

@media only screen and (max-width:768px) {
  .c0 {
    padding: 12px;
  }
}

@media only screen and (max-width:768px) {
  .c1 {
    margin-top: 0px;
  }
}

@media only screen and (max-width:768px) {
  .c1 {
    font-size: 26px;
    line-height: 32px;
    max-width: 624px;
  }
}

@media screen and (max-width:599px) {
  .c12 {
    display: none;
  }
}

@media screen and (max-width:599px) {
  .c13 {
    display: none;
  }
}

<div
  class="c0"
>
  <h1
    class="c1"
  >
    common.validators
  </h1>
  <div
    class="c2"
  >
    <div
      class="c3"
    >
      <div
        class="c4 rdt_Table"
        role="table"
      >
        <div
          class="c5 rdt_TableHead"
          role="rowgroup"
        >
          <div
            class="c6 rdt_TableHeadRow"
            role="row"
          >
            <div
              class="c7 c8 rdt_TableCol"
            />
            <div
              class="c7 c8 rdt_TableCol"
            />
            <div
              class="c7 c9 rdt_TableCol"
            >
              <div
                class="c10 rdt_TableCol_Sortable"
                id="column-name"
                role="columnheader"
                tabindex="0"
              >
                <div>
                  validator.name
                </div>
                <span
                  class="asc __rdt_custom_sort_icon__"
                >
                  <svg
                    aria-label="Down"
                    class="c11"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="m2 8.35 10.173 9.823L21.997 8"
                      fill="none"
                      stroke="#000"
                      stroke-width="2"
                    />
                  </svg>
                </span>
              </div>
            </div>
            <div
              class="c7 c12 rdt_TableCol"
            >
              <div
                class="c10 rdt_TableCol_Sortable"
                id="column-escrow"
                role="columnheader"
                tabindex="0"
              >
                <div>
                  validator.escrow
                </div>
                <span
                  class="asc __rdt_custom_sort_icon__"
                >
                  <svg
                    aria-label="Down"
                    class="c11"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="m2 8.35 10.173 9.823L21.997 8"
                      fill="none"
                      stroke="#000"
                      stroke-width="2"
                    />
                  </svg>
                </span>
              </div>
            </div>
            <div
              class="c7 c13 rdt_TableCol"
            >
              <div
                class="c10 rdt_TableCol_Sortable"
                id="column-fee"
                role="columnheader"
                tabindex="0"
              >
                <div>
                  validator.fee
                </div>
                <span
                  class="asc __rdt_custom_sort_icon__"
                >
                  <svg
                    aria-label="Down"
                    class="c11"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="m2 8.35 10.173 9.823L21.997 8"
                      fill="none"
                      stroke="#000"
                      stroke-width="2"
                    />
                  </svg>
                </span>
              </div>
            </div>
          </div>
        </div>
        <div
          class="c14 rdt_TableBody"
          offset="250px"
          role="rowgroup"
        >
          <div
            class="c15 rdt_TableRow"
            id="row-oasis1qpc4ze5zzq3aa5mu5ttu4ku4ctp5t6x0asemymfz"
            role="row"
          >
            <div
              class="c16 c8 c17 rdt_TableCell"
              data-tag="allowRowEvents"
              id="cell-icon-oasis1qpc4ze5zzq3aa5mu5ttu4ku4ctp5t6x0asemymfz"
              role="gridcell"
            />
            <div
              class="c16 c8 c17 rdt_TableCell"
              data-tag="allowRowEvents"
              id="cell-status-oasis1qpc4ze5zzq3aa5mu5ttu4ku4ctp5t6x0asemymfz"
              role="gridcell"
            >
              <svg
                aria-label="Status is okay"
                class="c18"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 22c5.523 0 10-4.477 10-10S17.523 2 12 2 2 6.477 2 12s4.477 10 10 10zM7 12l4 3 5-7"
                  fill="none"
                  stroke="#000"
                  stroke-width="2"
                />
              </svg>
            </div>
            <div
              class="c16 c9 c17 rdt_TableCell"
              data-tag="allowRowEvents"
              id="cell-name-oasis1qpc4ze5zzq3aa5mu5ttu4ku4ctp5t6x0asemymfz"
              role="gridcell"
            >
              test-validator1
            </div>
            <div
              class="c16 c12 c17 rdt_TableCell"
              data-tag="allowRowEvents"
              id="cell-escrow-oasis1qpc4ze5zzq3aa5mu5ttu4ku4ctp5t6x0asemymfz"
              role="gridcell"
            >
              <span>
                0
                <span
                  class="c19"
                >
                  ROSE
                </span>
              </span>
            </div>
            <div
              class="c16 c13 c17 rdt_TableCell"
              data-tag="allowRowEvents"
              id="cell-fee-oasis1qpc4ze5zzq3aa5mu5ttu4ku4ctp5t6x0asemymfz"
              role="gridcell"
            >
              10%
            </div>
          </div>
          <div
            class="c15 rdt_TableRow"
            id="row-oasis1qzyqaxestzlum26e2vdgvkerm6d9qgdp7gh2pxqe"
            role="row"
          >
            <div
              class="c16 c8 c17 rdt_TableCell"
              data-tag="allowRowEvents"
              id="cell-icon-oasis1qzyqaxestzlum26e2vdgvkerm6d9qgdp7gh2pxqe"
              role="gridcell"
            />
            <div
              class="c16 c8 c17 rdt_TableCell"
              data-tag="allowRowEvents"
              id="cell-status-oasis1qzyqaxestzlum26e2vdgvkerm6d9qgdp7gh2pxqe"
              role="gridcell"
            >
              <svg
                aria-label="Status is critical"
                class="c20"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12.703 2.703a.99.99 0 0 0-1.406 0l-8.594 8.594a.99.99 0 0 0 0 1.406l8.594 8.594a.99.99 0 0 0 1.406 0l8.594-8.594a.99.99 0 0 0 0-1.406l-8.594-8.594zM8.983 14.7 14.7 8.983m-5.717 0L14.7 14.7"
                  fill="none"
                  stroke="#000"
                  stroke-width="2"
                />
              </svg>
            </div>
            <div
              class="c16 c9 c17 rdt_TableCell"
              data-tag="allowRowEvents"
              id="cell-name-oasis1qzyqaxestzlum26e2vdgvkerm6d9qgdp7gh2pxqe"
              role="gridcell"
            >
              test-validator2
            </div>
            <div
              class="c16 c12 c17 rdt_TableCell"
              data-tag="allowRowEvents"
              id="cell-escrow-oasis1qzyqaxestzlum26e2vdgvkerm6d9qgdp7gh2pxqe"
              role="gridcell"
            >
              <span>
                0
                <span
                  class="c19"
                >
                  ROSE
                </span>
              </span>
            </div>
            <div
              class="c16 c13 c17 rdt_TableCell"
              data-tag="allowRowEvents"
              id="cell-fee-oasis1qzyqaxestzlum26e2vdgvkerm6d9qgdp7gh2pxqe"
              role="gridcell"
            >
              20%
            </div>
          </div>
          <div
            class="c15 rdt_TableRow"
            id="row-oasis1qrfe9n26nq3t6vc9hlu9gnupwf4rm6wr0uglh3r7"
            role="row"
          >
            <div
              class="c16 c8 c17 rdt_TableCell"
              data-tag="allowRowEvents"
              id="cell-icon-oasis1qrfe9n26nq3t6vc9hlu9gnupwf4rm6wr0uglh3r7"
              role="gridcell"
            />
            <div
              class="c16 c8 c17 rdt_TableCell"
              data-tag="allowRowEvents"
              id="cell-status-oasis1qrfe9n26nq3t6vc9hlu9gnupwf4rm6wr0uglh3r7"
              role="gridcell"
            >
              <svg
                aria-label="Status is unknown"
                class="c20"
                viewBox="0 0 24 24"
              >
                <path
                  d="M2 3.99C2 2.892 2.898 2 3.99 2h16.02C21.108 2 22 2.898 22 3.99v16.02c0 1.099-.898 1.99-1.99 1.99H3.99A1.995 1.995 0 0 1 2 20.01V3.99zM12 15v-1c0-1 0-1.5 1-2s2-1 2-2.5c0-1-1-2.5-3-2.5s-3 1.264-3 3m3 6v2"
                  fill="none"
                  stroke="#000"
                  stroke-width="2"
                />
              </svg>
            </div>
            <div
              class="c16 c9 c17 rdt_TableCell"
              data-tag="allowRowEvents"
              id="cell-name-oasis1qrfe9n26nq3t6vc9hlu9gnupwf4rm6wr0uglh3r7"
              role="gridcell"
            >
              test-validator3
            </div>
            <div
              class="c16 c12 c17 rdt_TableCell"
              data-tag="allowRowEvents"
              id="cell-escrow-oasis1qrfe9n26nq3t6vc9hlu9gnupwf4rm6wr0uglh3r7"
              role="gridcell"
            >
              <span>
                0
                <span
                  class="c19"
                >
                  ROSE
                </span>
              </span>
            </div>
            <div
              class="c16 c13 c17 rdt_TableCell"
              data-tag="allowRowEvents"
              id="cell-fee-oasis1qrfe9n26nq3t6vc9hlu9gnupwf4rm6wr0uglh3r7"
              role="gridcell"
            >
              20%
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
